packages/amplication-data-service-generator/src/server/resource/service/create-service.ts

```diff
@@ -375,6 +389,23 @@ async function createToManyRelationFile(
   return toManyFile;
 }
 
+function identityParentIdType(entity: Entity): namedTypes.Identifier {
+  const relatedEntityFiledId = entity.fields?.find(
+    (relatedEntityField) => relatedEntityField.dataType === EnumDataType.Id
```

Comment:
By doing this, you will always find the ID field of the entity and not the ID field of the related entity.

Reply:
You're right. I started with the related entity, but I tested it and it didn't work. I got the same error.

Reply:
I forgot to rename the variables